lib/sync-srv.js

```diff
@@ -297,33 +297,47 @@ function processPending(dataset_id, dataset, params, cb) {
           syncUtil.doLog(dataset_id, 'verbose', ' READ for UPDATE Success', params);
           syncUtil.doLog(dataset_id, 'silly', 'READ for UPDATE Data : \n' + util.inspect(data), params);
 
-          var preHash = syncUtil.generateHash(pre);
-          var dataHash = syncUtil.generateHash(data);
-
-          syncUtil.doLog(dataset_id, 'verbose', 'UPDATE Hash Check ' + uid + ' (client :: dataStore) = ' + preHash + ' :: ' + dataHash, params);
-
-          if (preHash === dataHash) {
-            DataSetModel.doUpdateHandler(dataset_id, uid, post, function (err) {
-              if (err) {
-                syncUtil.doLog(dataset_id, 'warn', 'UPDATE Failed - uid=' + uid + ' : err = ' + err, params);
-                return addUpdate("failed", "update", hash, uid, err, itemCallback);
+          //var preHash = syncUtil.generateHash(pre);
+          //var dataHash = syncUtil.generateHash(data);
+
+          //console.log('update action - sync : ' + preHash);
+          //console.log('update action - sync : ' + dataHash);
+
+          syncUtil.asyncGenerateHash(pre, function(err, preHash) {
+            syncUtil.asyncGenerateHash(data, function(err, dataHash) {
+              console.log('update action - async : ' + preHash);
+              console.log('update action - async : ' + dataHash);
+
+              syncUtil.doLog(dataset_id, 'verbose', 'UPDATE Hash Check ' + uid + ' (client :: dataStore) = ' + preHash + ' :: ' + dataHash, params);
+
+              if (preHash === dataHash) {
+                DataSetModel.doUpdateHandler(dataset_id, uid, post, function (err) {
+                  if (err) {
+                    syncUtil.doLog(dataset_id, 'warn', 'UPDATE Failed - uid=' + uid + ' : err = ' + err, params);
+                    return addUpdate("failed", "update", hash, uid, err, itemCallback);
+                  }
+                  syncUtil.doLog(dataset_id, 'info', 'UPDATE Success - uid=' + uid + ' : hash = ' + hash, params);
+                  return addUpdate("applied", "update", hash, uid, '', itemCallback);
+                }, meta_data);
+              } else {
+                //var postHash = syncUtil.generateHash(post);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)